From a614813622ff43d537be1d540f4baccee58f05a9 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 27 Feb 2014 16:59:43 +0000
Subject: [PATCH 169/351] extcon: arizona: Move measured hp_impedance to
 arizona

The switch driver was already updated, however the extcon driver is not.
This means that whilst running with the extcon driver the value in the
arizona structure remains at zero and is treated as a short preventing
headphone output.

Change-Id: Id303054d1fa00483291ea9a6d35c75be190ceea7
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/extcon/extcon-arizona.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/extcon/extcon-arizona.c b/drivers/extcon/extcon-arizona.c
index e82ae15..a0c572e 100644
--- a/drivers/extcon/extcon-arizona.c
+++ b/drivers/extcon/extcon-arizona.c
@@ -70,7 +70,6 @@ struct arizona_extcon_info {
 	struct input_dev *input;
 
 	u16 last_jackdet;
-	u32 hp_impedance;
 
 	int micd_mode;
 	const struct arizona_micd_config *micd_modes;
@@ -495,7 +494,7 @@ static int arizona_hpdet_read(struct arizona_extcon_info *info)
 		}
 	}
 
-	info->hp_impedance = val;
+	arizona->hp_impedance = val;
 	dev_dbg(arizona->dev, "HP impedance %d ohms\n", val);
 	return val;
 }
@@ -1129,7 +1128,7 @@ static irqreturn_t arizona_jackdet(int irq, void *data)
 		info->mic = false;
 		info->hpdet_done = false;
 		info->hpdet_retried = false;
-		info->hp_impedance = 0;
+		arizona->hp_impedance = 0;
 
 		for (i = 0; i < info->num_micd_ranges; i++)
 			input_report_key(info->input,
@@ -1243,7 +1242,7 @@ static ssize_t arizona_extcon_show(struct device *dev,
 	struct platform_device *pdev = to_platform_device(dev);
 	struct arizona_extcon_info *info = platform_get_drvdata(pdev);
 
-	return scnprintf(buf, PAGE_SIZE, "%d\n", info->hp_impedance);
+	return scnprintf(buf, PAGE_SIZE, "%d\n", info->arizona->hp_impedance);
 }
 
 static int arizona_extcon_probe(struct platform_device *pdev)
-- 
1.7.9.5

